core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/join/JoinHandler.java

```diff
@@ -64,21 +71,16 @@
                         context,
                         store);
 
-        final List leftList;
-        final List rightList;
+        final Iterable limitedLeftIterable;
+        final Iterable limitedRightIterable;
 
         try {
-            leftList = Lists.newArrayList(new LimitedCloseableIterable(operation.getInput(), 0, limit, false));
-            rightList = Lists.newArrayList(new LimitedCloseableIterable(rightIterable, 0, limit, false));
+            limitedLeftIterable = new LimitedCloseableIterable(operation.getInput(), 0, limit, false);
+            limitedRightIterable = new LimitedCloseableIterable(rightIterable, 0, limit, false);
+            return joinFunction.join(limitedLeftIterable, limitedRightIterable, operation.getMatchMethod(), matchKey, operation.isFlatten());
```

Comment:
This line can come out of the try block.

Reply:
I had to do this to make `shouldThrowExceptionWhenInputIsMoreThanLimit()` pass since the iterables aren't evaluated for size until they pass into the JoinFunction.